src/replayq.erl

```diff
@@ -226,13 +228,21 @@ append(#{config := #{seg_bytes := BytesLimit, dir := Dir} = Config,
 
 %% @doc pop out at least one item from the queue.
 %% volume limited by `bytes_limit' and `count_limit'.
--spec pop(q(), #{bytes_limit => bytes(), count_limit => count()}) ->
+-spec pop(q(),
+          #{
+            bytes_limit => bytes(),
+            count_limit => count(),
+            stop_before => stop_function(),
+            stop_before_input_accumulator => term()
+           }) ->
         {q(), ack_ref(), [item()]}.
 pop(Q, Opts) ->
   Bytes = maps:get(bytes_limit, Opts, ?DEFAULT_POP_BYTES_LIMIT),
   Count = maps:get(count_limit, Opts, ?DEFAULT_POP_COUNT_LIMIT),
+  StopFun = maps:get(stop_before, Opts, fun default_stop_function/2),
```

Comment:
export this function and use fun `?MODULE:default_stop_function/2`

Reply:
I guess this should be exported so that hot upgrade can work?

Reply:
Fixed